metrics.go

```diff
@@ -146,36 +139,43 @@ func (m *metrics) registerInstance() {
 
 func (m *metrics) sendMetrics() {
 	m.bucketMu.Lock()
-	if m.bucket.IsEmpty() {
-		m.resetBucket()
-		m.startTimer()
-		m.bucketMu.Unlock()
+	bucket := m.resetBucket()
+	m.bucketMu.Unlock()
+	if bucket.IsEmpty() {
 		return
 	}
-	payload := m.getPayload()
-	m.bucketMu.Unlock()
+	bucket.Stop = time.Now()
+	payload := MetricsData{
+		AppName:    m.options.appName,
+		InstanceID: m.options.instanceId,
+		Bucket:     bucket,
+	}
 
 	u, _ := m.options.url.Parse("./client/metrics")
-	m.startTimer()
 	resp, err := m.doPost(u, payload)
-
 	if err != nil {
 		m.err(err)
 		return
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode == http.StatusNotFound {
```

Comment:
What happens if the server deliberately does not handle metrics? Will we then accumulate metrics for-ever? Could that end up being a memory leak? 

From a user-perspective they should probably use the client with `disableMetrics=true` option set, maybe we should inform the user in the error message?

Reply:
No memory leak since we only ever store two `int32` in the bucket, but I guess theoretically it could overflow to be negative numbers at some point. Hopefully the server can handle those, but this is an edge case I think.

We will broadcast a warning every time the metrics fail to upload so the user does get informed to some extent.